src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,51 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $calendar-width;
+  gap: $box-gap;
+
+  &__cell {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    box-sizing: border-box;
+    width: $box-width;
+    height: $box-width;
+    background-color: $box-color;
+    border: 1px solid #eaeaea;
+
+    &--top-style {
+      border: none;
+    }
+  }
+
+  @each $day, $number in $days {
+    &--start-day--#{$day} .calendar__cell:first-child {
+      margin-inline-start: ($box-width * ($number - 1)) + ($number * $box-gap - 1);
+    }
+  }
+
+  @for $amount from 28 through 31 {
+    &--month-length--#{$amount} {
+      :nth-child(n + #{$amount + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  & .calendar__cell {
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: "#{$i}";
+      }
+    }
+
+    &:hover {
+      background-color: $color-secondary;
+      cursor: pointer;
+      transform: translateY(-5px);
+      transition-duration: 0.5s;
```

Comment:
move the transition properties to the main class not inside :hover
currently you don't have a transition back